### PR TITLE
Move cgmanifest.json

### LINKED ISSUE #125: cgmanifest.json: Move to a subdirectory
This is a machine-readable file for Microsoft-internal purposes. It must remain in our tree, but there's no reason for it to be at the top level, where it pushes the readme down. Moving it to the `stl` subdirectory would be an improvement (as it covers the code within). The tools that read this don't care about its location at all.

### PR DESCRIPTION
**Pull request by barcharcraz**
Please note that acceptance of community PRs will be delayed while we are
bringing our test and CI systems online. For more information, see the
README.md.

# Description
Move CGmanifest to a more appropriate location. Fixes #125.

mirror of internal PR https://devdiv.visualstudio.com/DevDiv/_git/msvc/pullrequest/204398?_a=overview

# Checklist:

- [x] I understand README.md.
- [x] If this is a feature addition, that feature has been voted into the C++
  Working Draft.
- [x] Identifiers in any product code changes are properly `_Ugly` as per
  https://eel.is/c++draft/lex.name#3.1 .
- [x] Identifiers in test code changes are *not* `_Ugly`.
- [x] Test code includes the correct headers as per the Standard, not just
  what happens to compile.
- [x] The STL builds and test harnesses have passed (must be manually verified
  by an STL maintainer before CI is online, leave this unchecked for initial
  submission).
- [x] This change introduces no known ABI breaks (adding members, renaming
  members, adding virtual functions, changing whether a type is an aggregate or
  trivially copyable, etc.). If unsure, leave this box unchecked and ask a
  maintainer for help.

### CODE REVIEW

File: docs/cgmanifest.json [hunk 1]
```diff
@@ -8,6 +8,15 @@
                     "commitHash": "59661c3f883dfd39cef6dc8eaf2fcbaae53597e8"
                 }
             }
+        },
+        {
+            "component": {
+                "type": "git",
+                "git": {
+                    "repositoryUrl": "https://github.com/microsoft/STL.git",
```
Comment (CaseyCarter) [inline docs/cgmanifest.json]: Oh no, the singularity! AHHHHhhhhhhhhh......

**Comment by CaseyCarter** [comment]
> I capitalized your PR title, and edited the description to mention which GitHub issue this fixes. Looks good to me.

Be sure to edit these changes into the commit log, either when merging or now with a force-push*.


*Ideally, `--force-with-lease`, which is a bit safer if other folks may have pushed into the branch.

**Comment by StephanTLavavej** [comment]
I am still unclear as to what happens when we press "Squash and merge". Does this ever use the original commit? This is related to whether commits appear as "Verified" or not, and why not all of them in the history are.

**Comment by sylveon** [comment]
@StephanTLavavej Squash and merge combines all commits of the PR into one, and adds that commit to the target branch. The resulting squashed commit is shown as verified because it is signed with GitHub's GPG key. PRs which are merged using the rebase or merge commit strategy aren't signed by GitHub. Rebase merges can't be verified because they edit the commits to replay them on top of the target branch and therefore break the signature, but merge commit merges will show verified on the original commits if the author signed them with their GPG key because it does not edit the commits.

**Comment by BillyONeal** [comment]
@sylveon It can't be just that because #114 was squashed but still doesn't show 'Verified'

**Comment by CaseyCarter** [comment]
> @sylveon It can't be just that because #114 was squashed but still doesn't show 'Verified'

#114 squashed unverified commits: https://github.com/microsoft/STL/pull/114/commits, so it wouldn't be signed.

**Comment by BillyONeal** [comment]
> #114 squashed unverified commits: https://github.com/microsoft/STL/pull/114/commits, so it wouldn't be signed.

#134 squashed unverified comments and shows up as "verified".

**Comment by sylveon** [comment]
Then I'm not sure what's happening here.

**Comment by BillyONeal** [comment]
That makes several of us :)

**Pull request merged**